augur/clades.py

```diff
@@ -0,0 +1,89 @@
+import os, sys
+import numpy as np
+from Bio import SeqIO, SeqFeature, Seq, SeqRecord, Phylo
+from .utils import read_node_data, write_json
+from collections import defaultdict
+
+def read_in_clade_definitions(clade_file):
+    '''
+    Reads in tab-seperated file that defines clades by amino-acid.
+
+    Format:
+    clade	gene	site	alt
+    Clade_2	embC	940	S
+    Clade_2	Rv3463	192	K
+    Clade_3	Rv2209	432	I
+    '''
+    import pandas as pd
+
+    clades = defaultdict(lambda:defaultdict(list))
+
+    df = pd.read_csv(clade_file, sep='\t' if clade_file.endswith('.tsv') else ',')
+    for mi, m in df.iterrows():
+        clades[m.clade][m.gene].append((m.site,m.alt))
+
+    return clades
+
+
+def is_node_in_clade(clade_mutations, node_muts):
+    '''
+    Determines whether a node contains all mutations that define a clade
+    '''
+    isClade = False
+    for gene, muts in clade_mutations.items():
+        if (gene in node_muts and node_muts[gene] != []):
+            prs_muts = [(int(tu[1:-1]), tu[-1]) for tu in node_muts[gene]] #get mutations in right format

```

Comment:
It is hard to tell what is happening here based on the variable names and operations alone. Can you rename these variables (especially `prs_muts`) to be more descriptive and include an inline comment above this line with an example of what `tu` looks like?

Reply:
the comparison with empty list is not ideal -- I am always worried it might be comparing pointers or similar. The ` != []` isn't actually needed, empty lists evaluate to false.